shipyard_blueprints/bigquery/shipyard_bigquery/bigquery.py

```diff
@@ -20,26 +20,52 @@
 
 
 class BigQueryClient(GoogleDatabase):
-    def __init__(self, service_account: str) -> None:
+    def __init__(
+        self, service_account: Optional[str] = None, access_token: Optional[str] = None
+    ) -> None:
+        if not service_account and not access_token:
+            raise ValueError("Either service account or access token must be provided")
         self.service_account = service_account
-        super().__init__(service_account)
+        self.access_token = access_token
+        super().__init__(service_account, access_token=access_token)
 
     @property
     def json_creds(self):
-        return json.loads(self.service_account)
+        if self.service_account:
+            return json.loads(self.service_account)
+        return
 
     @property
     def credentials(self):
+        if self.access_token:
+            return credentials.Credentials(
+                self.access_token,
+                scopes=[
+                    "https://www.googleapis.com/auth/bigquery",
+                    "https://www.googleapis.com/auth/bigquery.insertdata",
+                    "https://www.googleapis.com/auth/devstorage.read_write",
+                ],
+            )
         return service_account.Credentials.from_service_account_info(self.json_creds)
 
     @property
```

Comment:
is this property used/needed?

if not I would wrap this in a try clause then log a warning if the email is unable to be retrieved so that this doesn't throw a confusing error for the end user

Reply:
It is needed, particularly in the credential test and some of the blueprints